src/identity/manager.rs

```diff
@@ -296,10 +310,14 @@ impl Worker {
                     }
                     let (state, refresh_at) = match res {
                         Err(err) => {
-                            let refresh_at = Instant::now() + CERT_REFRESH_FAILURE_RETRY_DELAY;
+                            // Use the next backoff to determine when to retry the fetch and default to the constant value if the backoff has been reset.
+                            let refresh_at = Instant::now() + backoff.next_backoff().unwrap_or(CERT_REFRESH_FAILURE_RETRY_DELAY_INITIAL_INTERVAL);
```

Comment:
Hmm. I am a little worried about this if I understand right. Is this the behavior:

1. fetch cert1. Fails, enqueue with 500ms wait
2. fetch cert2. Fails, enqueue with 1s wait
...
3. fetch cert100. Fails, enqueue with 5m wait
4. fetch cert101. success! reset backoff
5.  ...cert100 only retries in 5m...

Two possible concerns:
* Each new cert ignores the backoff
* When we start succeeding again, all of our previously failed certs are stuck for a while

WDYT?

I guess it really depends on if the issue is caused by CA connectivity in general or that specific certificate. Both are possible

Reply:
The current parameters (specifically, max_elapsed_time) dictate that the total amount of time for all retries can't be longer than 5 minutes. But any one individual retry interval has a max of 60 seconds.

So iiuc, worst case would make it so that something retries for 60 seconds until the total time of 5 minutes has elapsed, then next_backoff starts returning none.

But I'm a bit confused now with the thing you shared before about the Go equivalent, if we set it to max_interval when it hits None. Won't it basically retry for 60 seconds indefinitely?

Maybe I should add some logic here to error out and have specific errors for what you've described:
```
I guess it really depends on if the issue is caused by CA connectivity in general or that specific certificate. Both are possible
```

At some point we'd want to just give up entirely and stop retrying. I'll have to spend some more time trying to understand this logic though. It's a bit confusing to me right now.

Reply:
The code, AFAIK, does not stop once it hits the max. `.unwrap_or(CERT_REFRESH_FAILURE_RETRY_DELAY_INITIAL_INTERVAL)`